sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jdbc/JdbcConnectionString.java

```diff
@@ -0,0 +1,136 @@
+    private final String jdbcURL;
+    private final Map<String, String> properties = new HashMap<>();
+    private DatabaseType databaseType = null;
+
+    private JdbcConnectionString(String jdbcURL) {
+        this.jdbcURL = jdbcURL;
+    }
+
+    private void resolveSegments() {
+        if (!StringUtils.hasText(this.jdbcURL)) {
+            LOGGER.warn("'connectionString' doesn't have text.");
+            throw new IllegalArgumentException(String.format(INVALID_CONNECTION_STRING_FORMAT, this.jdbcURL));
+        }
+
+        Optional<DatabaseType> optionalDatabaseType = Arrays.stream(DatabaseType.values())
+                                                            .filter(databaseType -> this.jdbcURL.startsWith(databaseType.getSchema()))
+                                                            .findAny();
+        this.databaseType = optionalDatabaseType.orElseThrow(() -> new AzureUnsupportedDatabaseTypeException(String.format(UNSUPPORTED_DATABASE_TYPE_STRING_FORMAT, this.jdbcURL)));
+
+        int pathQueryDelimiterIndex = this.jdbcURL.indexOf(this.databaseType.getPathQueryDelimiter());
+
+        if (pathQueryDelimiterIndex < 0) {
+            return;
+        }
+
+        String hostInfo = this.jdbcURL.substring(databaseType.getSchema().length() + 3, pathQueryDelimiterIndex);
+        String[] hostInfoArray = hostInfo.split(":");
+        if (hostInfoArray.length == 2) {
+            this.properties.put("servername", hostInfoArray[0]);
+            this.properties.put("port", hostInfoArray[1]);
+        } else {
+            this.properties.put("port", hostInfo);
+        }
+
+        String properties = this.jdbcURL.substring(pathQueryDelimiterIndex + 1);
+
+        final String[] tokenValuePairs = properties.split(this.databaseType.getQueryDelimiter());
+
+        for (String tokenValuePair : tokenValuePairs) {
+            final String[] pair = tokenValuePair.split(TOKEN_VALUE_SEPARATOR, 2);
+            String key = pair[0];
+            if (!StringUtils.hasText(pair[0])) {
+                throw new IllegalArgumentException(String.format(INVALID_PROPERTY_PAIR_FORMAT, tokenValuePair));
+            }
+            if (pair.length < 2) {
+                this.properties.put(key, NONE_VALUE);
+            } else {
+                this.properties.put(key, pair[1]);
+            }
+        }
+    }
+
+    public String enhanceConnectionString(Map<String, String> enhancedProperties) {
+        if (enhancedProperties == null || enhancedProperties.isEmpty()) {
+            return this.jdbcURL;
+        }
+        LOGGER.debug("Trying to enhance url for {}", databaseType);
+
+        StringBuilder builder = new StringBuilder(this.jdbcURL);
+
+        if (!this.hasProperties()) {
+            builder.append(databaseType.getPathQueryDelimiter());
+        } else {
+            builder.append(databaseType.getQueryDelimiter());
+        }
+
+        for (Map.Entry<String, String> entry : enhancedProperties.entrySet()) {
```

Comment:
what happen if the jdbcURL user provided already have some property defined but with different value?

for example, in the URL, it already have `PROPERTY_NAME_MYSQL_SSL_MODE` with a different than `PROPERTY_VALUE_MYSQL_SSL_MODE` value?

Reply:
It will throw IllegalArgumentException("Inconsistent property detected").